src/IO/S3/ProviderType.h

```diff
@@ -19,9 +19,13 @@ enum class ProviderType : uint8_t
 
 std::string_view toString(ProviderType provider_type);
 
-bool supportsMultiPartCopy(ProviderType provider_type);
+enum class ApiMode : uint8_t
+{
+    AWS,
+    GCS
```

Comment:
Can you please provide some comments about the enum and its constants?

Reply:
done